fuzztest/core/client_channel/retry_state_test.cc

```diff
@@ -0,0 +1,389 @@
+        return MakePolicy(absl::StrCat(
+            "{\"maxAttempts\":", max_attempts, ",\"initialBackoff\":\"",
+            Duration::Milliseconds(initial_backoff).ToJsonString(),
+            "\",\"maxBackoff\":\"",
+            Duration::Milliseconds(max_backoff).ToJsonString(),
+            "\",\"backoffMultiplier\":", backoff_multiplier,
+            ",\"retryableStatusCodes\":[",
+            absl::StrJoin(retryable_status_codes, ",",
+                          [](std::string* out, grpc_status_code c) {
+                            absl::StrAppend(
+                                out, "\"", grpc_status_code_to_string(c), "\"");
+                          }),
+            "]",
+            per_attempt_recv_timeout.has_value()
+                ? absl::StrCat(",\"perAttemptRecvTimeout\":\"",
+                               Duration::Milliseconds(*per_attempt_recv_timeout)
+                                   .ToJsonString(),
+                               "\"")
+                : "",
+            "}"));
+      },
+      InRange(2, 5), InRange(1, 100000), InRange(1, 100000),
+      InRange(1e-12, 10.0), VectorOf(AnyStatus()),
+      VariantOf(InRange<uint32_t>(1, 100000), AnyStatus()));
+}
+
+// Domain including valid retry configurations that only retry particular status
+// codes
+auto RetryMethodConfigWithRetryableStatusCodes(
+    std::vector<grpc_status_code> retryable_status_codes) {
+  return fuzztest::Map(
+      [retryable_status_codes](uint32_t max_attempts, uint32_t initial_backoff,
+                               uint32_t max_backoff,
+                               double backoff_multiplier) {
+        return MakePolicy(absl::StrCat(
+            "{\"maxAttempts\":", max_attempts, ",\"initialBackoff\":\"",
+            Duration::Milliseconds(initial_backoff).ToJsonString(),
+            "\",\"maxBackoff\":\"",
+            Duration::Milliseconds(max_backoff).ToJsonString(),
+            "\",\"backoffMultiplier\":", backoff_multiplier,
+            ",\"retryableStatusCodes\":[",
+            absl::StrJoin(retryable_status_codes, ",",
+                          [](std::string* out, grpc_status_code c) {
+                            absl::StrAppend(
+                                out, "\"", grpc_status_code_to_string(c), "\"");
+                          }),
+            "]}"));
+      },
+      InRange(2, 5), InRange(1, 100000), InRange(1, 100000),
+      InRange(1e-12, 10.0));
+}
+
+void Printable(absl::optional<internal::RetryMethodConfig> policy,
+               RefCountedPtr<internal::ServerRetryThrottleData> throttle_data) {
+  RetryState retry_state(policy.has_value() ? &*policy : nullptr,
+                         throttle_data);
+  std::ignore = absl::StrCat(retry_state);
+}
+FUZZ_TEST(MyTestSuite, Printable)
+    .WithDomains(OptionalOf(AnyRetryMethodConfig()), AnyServerThrottleData());
+
+void NoPolicyNeverRetries(std::vector<ServerMetadataHandle> md,
```

Comment:
This will probably need to change once you implement transparent retries.

Reply:
ack... though the current "AnyServerMetadata" domain only includes metadatum that don't have the not-seen-on-wire bits